internal/service/appstream/stack_user_association.go

```diff
@@ -0,0 +1,152 @@
+package appstream
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/appstream"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func ResourceUserStackAssociation() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceUserStackAssociationCreate,
+		ReadWithoutTimeout:   resourceUserStackAssociationRead,
+		DeleteWithoutTimeout: resourceUserStackAssociationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"authentication_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice(appstream.AuthenticationType_Values(), false),
+			},
+			"stack_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 128),
+			},
+			"user_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
```

Comment:
The parameter `send_email_notification` should be restored here, so that the email notification can be controlled. In https://github.com/hashicorp/terraform-provider-aws/pull/21485#discussion_r745172069, I meant that it should not be `ForceNew` and that changes should be ignored, since it only has an effect at creation time